### PR TITLE
Version Control Updates and Bug Fixes

### LINKED ISSUE #562: Bug: Repository Content Not Showing On Initial Showing
### Description

The repository tab in the version control navigator doesn't show all expanded items on the initial showcase

### To Reproduce

1. Go to Version Control in Project Navigator
2. Select the repository tab
3. Notice issue

### Expected behavior

It should show the expanded items correctly on the initial showing

### Version information

Aurora Editor: 1.0
Commit: Latest
MacOS: 14.1
Xcode: 15.1

### Additional context

_No response_

### LINKED ISSUE #668: Source Control: Uncaught Exception being thrown
### Description

Sometimes an unknown exception is being thrown in the GitShell class:

```
Thread 1: signal SIGABRT
An abort signal terminated the process. Such crashes often happen because of an uncaught exception or unrecoverable error or calling the abort() function.
```

### To Reproduce

Open any Source Control functionality

### Expected behavior

Not to crash

### Version information

Aurora Editor: 0.1
Commit: [e.g. d06fb30e72634048395b5dbcdbc756b7f8ecc2f0]
MacOS: 15 beta
Xcode: 16 beta

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nanashili**
## Summary
This commit introduces a series of updates and improvements to the version control system.

*  The branch selector now uses the new version control kit API, displaying current, recent, and other branches sorted by the date of last usage.
* A centralized location for observing command git data has been established.
* The project navigator now shows a pull icon next to files that have been changed on the upstream/origin branch.
* Fixed a bug causing the editor to crash due to the inability to fetch the project name.
* Resolved a UI issue where the inspector icon was showing in front of the divider instead of behind it.

## Changes Made

* Updated the branch selector to utilize the new version control kit API.
* Implemented a central observed location for command git data.
* Added pull icons to the project navigator for files changed on the upstream/origin branch.
* Fixed a bug causing the editor to crash when the project name could not be fetched.
* Corrected the UI issue with the inspector icon’s positioning relative to the divider.

## TODO

N/A

## Motivation

The primary goal of this PR is to enhance the version control experience by making branch selection more intuitive and providing visual indicators for upstream changes. Additionally, this PR addresses critical bugs and UI issues to improve overall stability and user experience.

## Testing

* Tested the branch selector functionality with the new API to ensure proper sorting and display of branches.
* Verified the central observation of command git data.
* Checked the project navigator for accurate display of pull icons next to changed files.
* Confirmed the editor no longer crashes when the project name cannot be fetched.
* Tested the UI fix for the inspector icon to ensure it appears correctly behind the divider.

## Screenshots/GIFs

N/A

## Checklist

Please ensure all of the following are completed before submitting the PR:

- [X] Code has been tested and verified.
- [X] Documentation has been updated to reflect the changes.
- [X] All tests pass successfully.
- [X] Code follows the project's coding guidelines and style.
- [X] The branch is up-to-date with the latest changes from the main branch.

## Related Issues

This PR addresses the following issue(s): 
https://github.com/AuroraEditor/AuroraEditor/issues/668
https://github.com/AuroraEditor/AuroraEditor/issues/562

## Additional Notes

N/A

### CODE REVIEW

**Comment by aurora-care-bear** [comment]
Thanks for submitting a pull request!
We will review it as soon as possible.

File: AuroraEditor/Core/InspectorSidebar/UI/History/HistoryItem.swift [hunk 1]
```diff
@@ -87,35 +87,35 @@ struct HistoryItem: View {
...
-                    Divider()
-                }
-                Button("Check Out \(commit.hash)...") {}
+//                if let commitRemoteURL = commit.?.absoluteString {
+//                    Button("View on \(commit.remoteString)...") {
+//                        let commitURL = "\(commitRemoteURL)/\(commit.commitHash)"
+//                        openCommit(URL(string: commitURL) ?? emptyURL)
+//                    }
+//                    Divider()
+//                }
```
Comment (0xWDG) [inline AuroraEditor/Core/InspectorSidebar/UI/History/HistoryItem.swift:117]: Is this deprecated / not used anymore?

Comment (nanashili) [reply AuroraEditor/Core/InspectorSidebar/UI/History/HistoryItem.swift:117]: No just commented out till i find a better way of doing it with the apis

**Review by 0xWDG — APPROVED** [review]

**Comment by 0xWDG** [comment]
@aurora-care-bear please approve this pr

**Review by aurora-care-bear — APPROVED** [review]
Approved on behalf of @0xWDG

**Comment by aurora-care-bear** [comment]
Hey @nanashili,

Thanks for your contribution.
Your PR has been merged!
Your name will appear on https://auroraeditor.com/#contributors in a few moments, if you have contributed to the editor.

Thanks again for your contribution!

Kind Regards,
Aurora Editor Team.

**Pull request merged**